doc/v5.0/man3/MPI_Status_set_elements.3
.\" -*- nroff -*-
.\" Copyright 2013 Los Alamos National Security, LLC. All rights reserved.
.\" Copyright 2010 Cisco Systems, Inc.  All rights reserved.
.\" Copyright 2006-2008 Sun Microsystems, Inc.
.\" Copyright (c) 1996 Thinking Machines
.\" Copyright (c) 2020      Google, LLC. All rights reserved.
.\" $COPYRIGHT$
.TH MPI_Status_set_elements 3 "Sep 30, 2021" "5.0.0rc1" "Open MPI"
.SH NAME
\fBMPI_Status_set_elements\fP, \fBMPI_Status_set_elements_x\fP \- Modifies opaque part of \fIstatus\fP to allow MPI_Get_elements to return \fIcount\fP.

.SH SYNTAX
.ft R
.SH C Syntax
.nf
#include <mpi.h>
int MPI_Status_set_elements(MPI_Status *\fIstatus\fP, MPI_Datatype \fIdatatype\fP, int \fIcount\fP)
int MPI_Status_set_elements_x(MPI_Status *\fIstatus\fP, MPI_Datatype \fIdatatype\fP, MPI_Count \fIcount\fP)

.fi
.SH Fortran Syntax
.nf
USE MPI
! or the older form: INCLUDE 'mpif.h'
MPI_STATUS_SET_ELEMENTS(\fISTATUS, DATATYPE, COUNT, IERROR\fP)
	INTEGER	\fISTATUS\fP(MPI_STATUS_SIZE), DATATYPE, COUNT, IERROR\fP
MPI_STATUS_SET_ELEMENTS_X(\fISTATUS, DATATYPE, COUNT, IERROR\fP)
	INTEGER	\fISTATUS\fP(MPI_STATUS_SIZE), DATATYPE\fP
        INTEGER(KIND=MPI_COUNT_KIND) \fICOUNT\fP
        INTEGER \fIIERROR\fP

.fi
.SH Fortran 2008 Syntax
.nf
USE mpi_f08
MPI_Status_set_elements(\fIstatus\fP, \fIdatatype\fP, \fIcount\fP, \fIierror\fP)
	TYPE(MPI_Status), INTENT(INOUT) :: \fIstatus\fP
	TYPE(MPI_Datatype), INTENT(IN) :: \fIdatatype\fP
	INTEGER, INTENT(IN) :: \fIcount\fP
	INTEGER, OPTIONAL, INTENT(OUT) :: \fIierror\fP
MPI_Status_set_elements_x(\fIstatus\fP, \fIdatatype\fP, \fIcount\fP, \fIierror\fP)
	TYPE(MPI_Status), INTENT(INOUT) :: \fIstatus\fP
	TYPE(MPI_Datatype), INTENT(IN) :: \fIdatatype\fP
	INTEGER(KIND = MPI_COUNT_KIND), INTENT(IN) :: \fIcount\fP
	INTEGER, OPTIONAL, INTENT(OUT) :: \fIierror\fP

.fi
.SH INPUT/OUTPUT PARAMETER
.ft R
.TP 1i
status
Status to associate with \fIcount\fP (status).

.SH INPUT PARAMETERS
.ft R
.TP 1i
datatype
Data type associated with \fIcount\fP (handle).
.TP 1i
count
Number of elements to associate with \fIstatus\fP (integer).

.SH OUTPUT PARAMETER
.ft R
.TP 1i
IERROR
Fortran only: Error status (integer).

.SH DESCRIPTION
.ft R
MPI_Status_set_elements modifies the opaque part of \fIstatus\fP so that a call to MPI_Get_elements or MPI_Get_elements_x will return \fIcount\fP. MPI_Get_count will return a compatible value.
.sp
A subsequent call to MPI_Get_count(\fIstatus, datatype, count\fP), to MPI_Get_elements(\fIstatus, datatype, count\fP), or to MPI_Get_elements_x(\fIstatus, datatype, count\fP) must use a data-type argument that has the same type signature as the data-type argument that was used in the call to MPI_Status_set_elements.

.SH NOTES
.ft R
Users are advised not to reuse the status fields for values other than those for which they were intended. Doing so may lead to unexpected results when using the status object. For example, calling MPI_Get_elements may cause an error if the value is out of range, or it may be impossible to detect such an error. The \fIextra_state\fP argument provided with a generalized request can be used to return information that does not logically belong in \fIstatus\fP. Furthermore, modifying the values in a status set internally by MPI, such as MPI_Recv, may lead to unpredictable results and is strongly discouraged.

.SH ERRORS
Almost all MPI routines return an error value; C routines as the value of the function and Fortran routines in the last argument.
.sp
Before the error value is returned, the current MPI error handler is
called. By default, this error handler aborts the MPI job, except for I/O function errors. The error handler may be changed with MPI_Comm_set_errhandler; the predefined error handler MPI_ERRORS_RETURN may be used to cause error values to be returned. Note that MPI does not guarantee that an MPI program can continue past an error.

.SH FORTRAN 77 NOTES
.ft R
The MPI standard prescribes portable Fortran syntax for the \fICOUNT\fP argument of MPI_Status_set_elements_x only for Fortran 90. FORTRAN 77 users may use the non-portable syntax
.sp
.nf
     INTEGER*MPI_COUNT_KIND \fICOUNT\fP
.sp
where MPI_COUNT_KIND is a constant defined in mpif.h and gives the length of the declared integer in bytes.


doc/v1.4/man3/MPI_Status_set_elements.3.php
<?php
$topdir = "../../..";
$title = "MPI_Status_set_elements(3) man page (version 1.4.5)";
$meta_desc = "Open MPI v1.4.5 man page: MPI_STATUS_SET_ELEMENTS(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
  <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_Status_set_elements</b> - Modifies opaque part of <i>status</i> to allow
<a href="../man3/MPI_Get_elements.3.php">MPI_Get_elements</a> to return <i>count</i>.
<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<h2><a name='sect2' href='#toc2'>C Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_Status_set_elements(MPI_Status *status, MPI_Datatype datatype,
int count)
</pre>
<h2><a name='sect3' href='#toc3'>Fortran Syntax</a></h2>
<br>
<pre>INCLUDE &rsquo;mpif.h&rsquo;
MPI_STATUS_SET_ELEMENTS(STATUS, DATATYPE, COUNT, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER<tt> </tt>&nbsp;<tt> </tt>&nbsp;STATUS(MPI_STATUS_SIZE), DATATYPE, COUNT, IERROR
</pre>
<h2><a name='sect4' href='#toc4'>C++ Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
void MPI::Status::Set_elements(const MPI::Datatype&amp; datatype, int count)
</pre>
<h2><a name='sect5' href='#toc5'>Input/Output Parameter</a></h2>

<dl>

<dt>status </dt>
<dd>Status to associate with <i>count</i> (status).
<p> </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Input
Parameters</a></h2>

<dl>

<dt>datatype </dt>
<dd>Data type associated with <i>count</i> (handle). </dd>

<dt>count </dt>
<dd>Number
of elements to associate with <i>status</i> (integer).
<p> </dd>
</dl>

<h2><a name='sect7' href='#toc7'>Output Parameter</a></h2>

<dl>

<dt>IERROR
</dt>
<dd>Fortran only: Error status (integer).
<p> </dd>
</dl>

<h2><a name='sect8' href='#toc8'>Description</a></h2>
MPI_Status_set_elements
modifies the opaque part of <i>status</i> so that a call to <a href="../man3/MPI_Get_elements.3.php">MPI_Get_elements</a> will
return <i>count</i>. <a href="../man3/MPI_Get_count.3.php">MPI_Get_count</a> will return a compatible value. <p>
A subsequent
call to <a href="../man3/MPI_Get_count.3.php">MPI_Get_count</a>(<i>status, datatype, count</i>) or to <a href="../man3/MPI_Get_elements.3.php">MPI_Get_elements</a>(<i>status,
datatype, count</i>) must use a data-type argument that has the same type signature
as the data-type argument that was used in the call to MPI_Status_set_elements.

<p>
<h2><a name='sect9' href='#toc9'>Notes</a></h2>
Users are advised not to reuse the status fields for values other
than those for which they were intended. Doing so may lead to unexpected
results when using the status object. For example, calling <a href="../man3/MPI_Get_elements.3.php">MPI_Get_elements</a>
may cause an error if the value is out of range, or it may be impossible
to detect such an error. The <i>extra_state</i> argument provided with a generalized
request can be used to return information that does not logically belong
in <i>status</i>. Furthermore, modifying the values in a status set internally
by MPI, such as <a href="../man3/MPI_Recv.3.php">MPI_Recv</a>, may lead to unpredictable results and is strongly
discouraged.
<p>
<h2><a name='sect10' href='#toc10'>Errors</a></h2>
Almost all MPI routines return an error value; C routines
as the value of the function and Fortran routines in the last argument.
C++ functions do not return errors. If the default error handler is set
to MPI::ERRORS_THROW_EXCEPTIONS, then on error the C++ exception mechanism
will be used to throw an MPI:Exception object. <p>
Before the error value is
returned, the current MPI error handler is called. By default, this error
handler aborts the MPI job, except for I/O function errors. The error handler
may be changed with <a href="../man3/MPI_Comm_set_errhandler.3.php">MPI_Comm_set_errhandler</a>; the predefined error handler
MPI_ERRORS_RETURN may be used to cause error values to be returned. Note
that MPI does not guarantee that an MPI program can continue past an error.

<p> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>C Syntax</a></li>
<li><a name='toc3' href='#sect3'>Fortran Syntax</a></li>
<li><a name='toc4' href='#sect4'>C++ Syntax</a></li>
<li><a name='toc5' href='#sect5'>Input/Output Parameter</a></li>
<li><a name='toc6' href='#sect6'>Input Parameters</a></li>
<li><a name='toc7' href='#sect7'>Output Parameter</a></li>
<li><a name='toc8' href='#sect8'>Description</a></li>
<li><a name='toc9' href='#sect9'>Notes</a></li>
<li><a name='toc10' href='#sect10'>Errors</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
